sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -358,20 +316,25 @@ public class EventProcessorClient : IAsyncDisposable
                                                 EventHubConnection connection,
                                                 EventProcessorClientOptions processorOptions)
         {
+            // TODO: we probably can remove this constructor and OwnsConnection property because the processor does not have
+            // a single connection anymore.  In fact, returning the same connection from the factory might result in undefined
```

Comment:
Because the base class manages the created connections by itself, and they might be closed if needed.

For example, we'll use the factory get a connection when creating a consumer for partition processing. If we lose ownership of the partition being processed, we'll stop the processing task, close its consumer and its connection.

Reply:
I needed to update the `EventProcessorManager` in the live tests to make them work again exactly because we were always using the same connection. In a non-live test scenario, I think we could still use this constructor, but we needed to make sure the connection's `CloseAsync` method is overridden to do nothing.

Reply:
Another property should be enough. I don't see a good reason for managing some connections but not all of them. `CreateConnection` does not take in any parameters, so the user wouldn't be able to differentiate them anyway.

If we really want to allow the users to be responsible for closing a connection, we also need a way to tell them when it's safe to be disposed. We would need to individually do this for every single connection so they can stop holding references that are not needed anymore.

However, I can't really understand why someone would need to close the connections themself. I can think of two possible scenarios:
- **The user wants to use the same connection after it's used by the Event Processor:** what is the benefit of doing this instead of spawning a new connection?
- **The user wants to use the same connection while the Event Processor is still using it:** if I recall correctly, we wanted to let the processor manage how many connections it uses and how to distribute them in order to improve performance. Letting the user use the connections for other stuff simultaneously may result in unexpected performance drops.

I know we are dealing with advanced users, but is the benefit of letting them manage the connections by themselves that great? I may be overlooking some scenarios you may be aware of.

Reply:
I'm thinking in terms of testing, not for actual library users.   But, it's highly likely that I'm overthinking things and the benefits don't outweigh the costs.